test/es-module/test-esm-dynamic-import-commonjs.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const assert = require('node:assert');
+const fs = require('node:fs/promises');
+
+tmpdir.refresh();
+
+const preloadURL = tmpdir.fileURL('preload.cjs');
+const commonjsURL = tmpdir.fileURL('commonjs-module-1.cjs');
+
+(async () => {
+
+  await Promise.all([fs.writeFile(preloadURL, ''), fs.writeFile(commonjsURL, '')]);
+
+  // Make sure that the CommonJS module lexer has been initialized.
+  // See https://github.com/nodejs/node/blob/v21.1.0/lib/internal/modules/esm/translators.js#L61-L81.
+  await import(preloadURL);
```

Comment:
Let's not create new files, it tends to make tests flaky
```suggestion
  // Make sure that the CommonJS module lexer has been initialized.
  // See https://github.com/nodejs/node/blob/v21.1.0/lib/internal/modules/esm/translators.js#L61-L81.
  await import(fixtures.fileURL('empty.js'));
```

Reply:
I've updated the unit tests to use fixtures and checked that tests are still failing when the fix is reverted: https://github.com/nodejs/node/pull/50465/commits/bc7dcaa270f9da2773ed4c48ec45e18c437cb759. Thanks for the suggestions!